serverless/serverless
# Make code coverage great again (return it back to 100% - ref: #3275)

## What did you implement:

Closes #3275



Implemented missing tests to make each line of code to be executed at least once. That is 100% code coverage. Also a few improvements in the tests and code.

## How did you implement it:



Added the missing tests.

## How can we verify it:



In order to verify - run tests and lint.
Here are the results on the PR code:

* [Travis CI](https://travis-ci.org/zerkella/serverless)
* [Coveralls](https://coveralls.io/github/zerkella/serverless)

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
